apps/SageAccountWeb/src/components/SignUpdatedTermsOfUsePage.tsx

```diff
@@ -37,6 +33,14 @@ export const SignUpdatedTermsOfUsePage = (
     },
   )
 
+  const redirectAfterSignOrSkip = () => {
+    const didRedirect = restoreLastPlace(history)
+    if (!didRedirect) {
+      // if we did not redirect to a page in this app, then look for the redirect cookie
+      processRedirectURLInOneSage()
+    }
```

Comment:
minor change: if we redirected you within OneSage (e.g. to your account settings page), react-router would handle it, so `processRedirectURLInOneSage` would run unexpectedly.

Updated `restoreLastPlace` to return a boolean if a redirect happened, and use it to decide whether to redirect using the cookie.